staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -42,11 +42,19 @@ import (
 type AdmissionOptions struct {
 	// RecommendedPluginOrder holds an ordered list of plugin names we recommend to use by default
 	RecommendedPluginOrder []string
-	// DefaultOffPlugins a list of plugin names that should be disabled by default
-	DefaultOffPlugins []string
-	PluginNames       []string
-	ConfigFile        string
-
+	// DefaultEnabledPlugins is a set of plugin names that is enabled by default
+	DefaultEnabledPlugins sets.String
+	// DEPRECATED flag, should use EnabledAdmissionPlugins and DisabledAdmissionPlugins.
+	// They are mutually exclusive, specify both will lead to an error.
+	PluginNames []string
```

Comment:
Only `kube-apiserver` uses this, right?  Move it out of the generic apiserver.

Reply:
I think genericapiserver can also use it, since admission is add to RecommendOptions.

Reply:
> I think genericapiserver can also use it, since admission is add to RecommendOptions.

That only happened in 1.10.  We don't want people to do this generally. They usually get it wrong.  If a server really needs the feature, it can add it, but we should not make it easy in the genericapiserver.  The genericapiserver can be opinionated and we (openshift/kube) have deep experience in this area.  Letting users specify it consistently results in bad configurations.

Reply:
ok, I have a doubt how it differentiate from enable-admission-plugins? Are they both kube-apiserver specific?

Reply:
Can we create a super type which embeds the normal AdmissionOptions, but adds the `PluginNames`. Then we don't offer that flag by default, but people (and kube-apiserver) can opt-in to that deprecated feature.

Reply:
> Can we create a super type which embeds the normal AdmissionOptions, but adds the PluginNames? Then we don't offer that flag by default, but people (and kube-apiserver) can opt-in to that deprecated feature.

How about we start that super-type in `pkg/kubeapiserver` first and if we encounter a strong reason to expose it we move it then.

Reply:
Sounds good.